src/workerd/api/global-scope.c++

```diff
@@ -471,16 +471,28 @@ kj::Promise<WorkerInterface::AlarmResult> ServiceWorkerGlobalScope::runAlarm(kj:
             !jsg::isDoNotLogException(desc) && context.isOutputGateBroken()) {
           LOG_NOSENTRY(ERROR, "output lock broke during alarm execution", actorId, e);
         } else if (context.isOutputGateBroken()) {
-          // We don't usually log these messages, but it's useful to know the real reason we failed
-          // to correctly investigate stuck alarms.
-          LOG_NOSENTRY(ERROR,
-              "output lock broke during alarm execution without an interesting error description",
-              actorId, e);
+          if (e.getType() == kj::Exception::Type::OVERLOADED) {
+            // This is a bit hackish, but we know for a fact that this specific exception detail is
+            // caused by user error, but not all overloaded exceptions are user fault. As such
+            // let's treat this special case.
+            // TODO(soon): Get a more robust way to process user errors comming from JSG exceptions.
+            if (e.getDescription() ==
```

Comment:
Why 2 `if`s when we really only care about both being true? Could the `if` above combine them with an `&&`?

Alternatively, on line 485 we could do an OR along with `overloaded && matchesString`?

Reply:
Sorry, ignore me. I didn't realize that's what was happening below :sweat_smile:.